CRM/Utils/Mail.php

```diff
@@ -649,12 +649,16 @@ public static function format($fields) {
    * If it's numeric, look up the display name and email of the corresponding
    * contact ID in RFC822 format.
    *
-   * @param string $from
+   * @param string|array $from
    *   civicrm_email.id or formatted "From address", eg. 12 or "Fred Bloggs" <fred@example.org>
+   *   or array containing 'name' and 'email' keys.
    * @return string
    *   The RFC822-formatted email header (display name + address)
    */
   public static function formatFromAddress($from) {
+    if (is_array($from)) {
+      return "\"{$from['name']}\" <{$from['email']}>";
```

Comment:
Should quotes or angular brackets be encoded or escaped here?

Reply:
This function returns an unescaped value. See a few lines down.